### PR TITLE
Redirect student to first incomplete quiz question

### PR DESCRIPTION
**Pull request by cmkoller**
Also:
- Rename a test from "student" to "user"
- Added a missing test to question#student_answer to return nil if there is no answer

### CODE REVIEW

File: spec/features/user_is_redirected_to_last_unanswered_question_spec.rb [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+  scenario "student returns to unanswered quiz" do
+    ActionDispatch::Request.any_instance.stub(:remote_ip).and_return(ENV['LAUNCH_ACADEMY_IP'])
+    sign_in(user)
+    visit quiz_path(quiz)
+    within(all('.new_answer_submission').first) { click_on 'X' }
+
+    visit quizzes_path
+
+    visit quiz_path(quiz)
```
Comment (radavis) [inline spec/features/user_is_redirected_to_last_unanswered_question_spec.rb]: To reproduce the error we are seeing with students:
1) Start the quiz
2) Answer a question
3) Navigate back to quizzes_path, and click on the quiz

So, here, instead of visiting the quiz path, click on the link. I know that the results should be same, but use the features of capybara to interact with the webapp as the user would.

File: spec/models/question_spec.rb [hunk 2]
```diff
@@ -7,14 +7,24 @@
...
+      expect(question.student_answer(user)).to eq(question.correct_answer)
+    end
+
+    it "returns nil if student has not answered" do
+      user = FactoryGirl.create(:user)
+      quiz = FactoryGirl.create(:quiz_with_questions)
+      question = quiz.questions.first
+
+      expect(question.student_answer(user)).to eq(nil)
+    end
```
Comment (radavis) [inline spec/models/question_spec.rb:28]: Nice work on these Model specs

File: spec/models/quiz_spec.rb [hunk 3]
```diff
@@ -62,4 +62,27 @@
...
+      second_question = FactoryGirl.create(:question_with_answers, quiz: quiz)
+
+      expect(quiz.first_unanswered_question(user)).to eq(second_question)
+    end
+
+    it "returns nil if student has answered all questions" do
+      first_question = quiz.questions.first
+      FactoryGirl.create(:answer_submission,
+      user: user,
+      answer: first_question.correct_answer
```
Comment (radavis) [inline spec/models/quiz_spec.rb]: I added launchcop to the gemfile. After you get your tests passing, and before you push your code to github, get into the habit of running `bundle exec launchcop` and fixing any errors you might find.

The message you will get from this line is: "spec/models/quiz_spec.rb:81:7: C: Align the parameters of a method call if they span more than one line." It should say to indent the lines like on 71, 72. Which is what the launchcop settings are set to. This seems like a rubocop bug.

**Comment by radavis** [comment]
Overall, looks good. Fix these issues and try it out on staging.
